xilem/src/view/sized_box.rs

```diff
@@ -209,3 +221,84 @@ struct BorderStyle {
     width: f64,
     color: Color,
 }
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct Padding {
+    top: f64,
+    trailing: f64,
+    bottom: f64,
+    leading: f64,
+}
+
+impl Padding {
+    pub const fn new(top: f64, trailing: f64, bottom: f64, leading: f64) -> Self {
+        Self {
+            top,
+            trailing,
+            bottom,
+            leading,
+        }
+    }
+
+    pub const ZERO: Padding = Padding::all(0.);
+
+    pub const fn all(padding: f64) -> Self {
+        Self::new(padding, padding, padding, padding)
+    }
+
+    pub const fn horizontal(padding: f64) -> Self {
+        Self::new(0., padding, 0., padding)
+    }
+
+    pub const fn vertical(padding: f64) -> Self {
+        Self::new(padding, 0., padding, 0.)
+    }
+
+    pub const fn top(padding: f64) -> Self {
+        Self::new(padding, 0., 0., 0.)
+    }
+
+    pub const fn trailing(padding: f64) -> Self {
+        Self::new(0., padding, 0., 0.)
+    }
+
+    pub const fn bottom(padding: f64) -> Self {
+        Self::new(0., 0., padding, 0.)
+    }
+
+    pub const fn leading(padding: f64) -> Self {
```

Comment:
So in an LTR context this is left, and a RTL context this is right?

We should document that clearly. Especially since we don't actually handle bidi yet...

Reply:
I have added a comment about it in the documentation of the struct attributes. Please let me know if that look good